search/includes/classes/class-queue.php

```diff
@@ -402,4 +426,23 @@ public function intercept_ep_sync_manager_indexing( $bail, $sync_manager, $index
 
 		return true;
 	}
+
+	/**
+	 * Hook after bulk indexing looking for errors. If there's an error with indexing some of the posts and the queue is enabled, 
+	 * queue all of the posts for indexing.
+	 *
+	 * @param {array} $document_ids IDs of the documents that were to be indexed
+	 * @param {string} $slug Indexable slug
+	 * @param {array|boolean} $return Elasticsearch response. False on error.
+	 * @return {bool} Whether anything was done
+	 */
+	public function action__ep_after_bulk_index( $document_ids, $slug, $return ) {
+		if ( false === $this->is_enabled() || ! is_array( $document_ids ) || 'post' !== $slug || false !== $return ) {
```

Comment:
I think we are ok to allow offloading of other types - the queue system supports it so _should_ be ok, unless you'd rather just keep it simple for now.

Reply:
I just want to keep it simple, explicit, and limited to what we've tested thoroughly.